File: library/src/main/java/com/google/maps/android/clustering/algo/ScreenBasedAlgorithmAdapter.java
/*
 * Copyright 2016 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.maps.android.clustering.algo;

import com.google.android.libraries.maps.model.CameraPosition;
import com.google.maps.android.clustering.Cluster;
import com.google.maps.android.clustering.ClusterItem;

import java.util.Collection;
import java.util.Set;

public class ScreenBasedAlgorithmAdapter<T extends ClusterItem> extends AbstractAlgorithm<T> implements ScreenBasedAlgorithm<T> {

    private Algorithm<T> mAlgorithm;

    public ScreenBasedAlgorithmAdapter(Algorithm<T> algorithm) {
        mAlgorithm = algorithm;
    }

    @Override
    public boolean shouldReclusterOnMapMovement() {
        return false;
    }

    @Override
    public boolean addItem(T item) {
        return mAlgorithm.addItem(item);
    }

    @Override
    public boolean addItems(Collection<T> items) {
        return mAlgorithm.addItems(items);
    }

    @Override
    public void clearItems() {
        mAlgorithm.clearItems();
    }

    @Override
    public boolean removeItem(T item) {
        return mAlgorithm.removeItem(item);
    }

    @Override
    public boolean removeItems(Collection<T> items) {
        return mAlgorithm.removeItems(items);
    }

    @Override
    public boolean updateItem(T item) {
        return mAlgorithm.updateItem(item);
    }

    @Override
    public Set<? extends Cluster<T>> getClusters(float zoom) {
        return mAlgorithm.getClusters(zoom);
    }

    @Override
    public Collection<T> getItems() {
        return mAlgorithm.getItems();
    }

    @Override
    public void setMaxDistanceBetweenClusteredItems(int maxDistance) {
        mAlgorithm.setMaxDistanceBetweenClusteredItems(maxDistance);
    }

    @Override
    public int getMaxDistanceBetweenClusteredItems() {
        return mAlgorithm.getMaxDistanceBetweenClusteredItems();
    }

    @Override
    public void onCameraChange(CameraPosition cameraPosition) {
        // stub
    }

}

File: clustering/src/main/java/com/google/maps/android/clustering/algo/ScreenBasedAlgorithmAdapter.kt
/*
 * Copyright 2016 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.maps.android.clustering.algo

import com.google.android.gms.maps.model.CameraPosition
import com.google.maps.android.clustering.Cluster
import com.google.maps.android.clustering.ClusterItem

class ScreenBasedAlgorithmAdapter<T : ClusterItem>(private val algorithm: Algorithm<T>) : AbstractAlgorithm<T>(), ScreenBasedAlgorithm<T> {

    override fun shouldReclusterOnMapMovement(): Boolean {
        return false
    }

    override fun addItem(item: T): Boolean {
        return algorithm.addItem(item)
    }

    override fun addItems(items: Collection<T>): Boolean {
        return algorithm.addItems(items)
    }

    override fun clearItems() {
        algorithm.clearItems()
    }

    override fun removeItem(item: T): Boolean {
        return algorithm.removeItem(item)
    }

    override fun removeItems(items: Collection<T>): Boolean {
        return algorithm.removeItems(items)
    }

    override fun updateItem(item: T): Boolean {
        return algorithm.updateItem(item)
    }

    override fun getClusters(zoom: Float): Set<Cluster<T>> {
        return algorithm.getClusters(zoom)
    }

    override val items: Collection<T>
        get() = algorithm.items

    override var maxDistanceBetweenClusteredItems: Int
        get() = algorithm.maxDistanceBetweenClusteredItems
        set(maxDistance) {
            algorithm.maxDistanceBetweenClusteredItems = maxDistance
        }

    override fun onCameraChange(position: CameraPosition) {
        // stub
    }
}
